src/common/playlist/playlist-manager.ts

```diff
@@ -365,6 +386,21 @@ class PlaylistManager {
         return this._player.loadMedia(this._mediaInfoList[index]).then((mediaConfig) => {
           this._playlist.updateItemSources(index, mediaConfig.sources);
           this._playlist.updateItemPlugins(index, mediaConfig.plugins);
+
+          let sessionId = this._player.sessionIdCache?.get(mediaConfig.sources.id);
+          if (!sessionId) {
+            sessionId = SessionIdGenerator.get();
+            this._player.sessionIdCache?.set(mediaConfig.sources.id, sessionId);
+          }
+
+          mediaConfig.sources.dash = mediaConfig.sources.dash.map((source) => {
```

Comment:
Is this only relevant for dash, seems that for now its usable for Dash but future wise its for all